java/src/test/java/com/saucelabs/saucebindings/integration/VisualTest.java

```diff
@@ -0,0 +1,116 @@
+import com.saucelabs.saucebindings.SauceSession;
+import com.saucelabs.saucebindings.SystemManager;
+import com.saucelabs.saucebindings.options.InvalidSauceOptionsArgumentException;
+import com.saucelabs.saucebindings.options.SauceOptions;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.openqa.selenium.remote.RemoteWebDriver;
+
+import java.util.List;
+import java.util.Map;
+
+import static org.junit.Assert.*;
+
+public class VisualTest {
+    private SauceSession session;
+    private RemoteWebDriver webDriver;
+
+    @Before
+    public void setup() {
+        System.setProperty("BUILD_NAME", "Sauce Bindings");
+    }
+
+    @After
+    public void cleanUp() {
+        if (session != null) {
+            session.stop(true);
+        }
+    }
+
+    @Test
+    public void usesVisualWhenOptionSet() {
+        SauceOptions options = SauceOptions.chrome().setName("Visual uses this name").setVisual().build();
+        session = new SauceSession(options);
+        webDriver = session.start();
+        session.visual().takeSnapshot("Snapshot necessary to for Visual test to pass");
+        assertNotNull(webDriver);
+        assertTrue(session.getIsVisualSession());
+    }
+
+    @Test
+    public void usesVisualWhenDataCenterSet() {
+        SauceOptions options = SauceOptions.chrome().setName("Visual uses this name").build();
+        session = new SauceSession(options);
+        session.setDataCenter(DataCenter.VISUAL);
+        webDriver = session.start();
+        session.visual().takeSnapshot("Snapshot necessary to for Visual test to pass");
+        assertNotNull(webDriver);
+        assertTrue(session.getIsVisualSession());
+    }
+
+    @Test
+    public void acceptsNameForInit() {
+        SauceOptions options = SauceOptions.chrome().setName("Visual does not use this name").setVisual().build();
+        session = new SauceSession(options);
+        webDriver = session.start();
+        session.visual().init("Visual uses this name instead");
+        session.visual().takeSnapshot("Snapshot necessary to for Visual test to pass");
+        Map results = session.visual().getResults();
+        Map snapshot = (Map) ((List) results.get("states")).get(0);
+        assertEquals("Visual uses this name instead", snapshot.get("groupName"));
+    }
+
+    @Test
+    public void takesSnapshot() {
+        SauceOptions options = SauceOptions.chrome().setName("Takes Sauce Snapshot").setVisual().build();
+        session = new SauceSession(options);
+        webDriver = session.start();
+        session.visual().takeSnapshot("Name of Snapshot");
+        Map results = session.visual().getResults();
+        Map snapshot = (Map) ((List) results.get("states")).get(0);
+        assertEquals("Name of Snapshot", snapshot.get("name"));
+    }
+
+    @Test
+    public void providesTestResults() {
+        SauceOptions options = SauceOptions.chrome().setName("Provides Test Results").setVisual().build();
```

Comment:
Yes, I had an implementation that supported this in a couple different ways... but then I thought that ~90% of our users already set a name in their tests, and I'd rather encourage the other 10% to do it as well, rather than add extra code.